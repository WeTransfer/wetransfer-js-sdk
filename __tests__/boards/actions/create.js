const routes = require('../../../src/config/routes');
const createAction = require('../../../src/boards/actions/create');
const WTError = require('../../../src/error');

describe('Create board action', () => {
  let create = null;
  const mocks = {};
  beforeEach(() => {
    mocks.request = { send: jest.fn() };
    mocks.request.send.mockReturnValue({
      id: 'random-hash',
      name: 'Little kittens',
      description: null,
      state: 'uploading',
      url: 'https://we.tl/s-random-hash',
      items: [],
    });

    create = createAction({
      routes,
      request: mocks.request,
    });
  });

  it('should create a new board request', async () => {
    const board = await create({
      name: 'WeTransfer SDK',
    });
    expect(board).toMatchSnapshot();
  });

  it('should throw an exception if the request fails', async () => {
    mocks.request.send.mockImplementation(() =>
      Promise.reject(new Error('Network error.'))
    );

    try {
      await create({
        name: 'WeTransfer SDK',
      });
      await authorize();
    } catch (error) {
      expect(error).toBeInstanceOf(WTError);
    }
  });
});
