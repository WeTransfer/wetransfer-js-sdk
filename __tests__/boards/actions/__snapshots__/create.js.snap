// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Create board action should create a new board request 1`] = `
RemoteBoard {
  "description": null,
  "id": "random-hash",
  "items": Array [],
  "name": "Little kittens",
  "state": "uploading",
  "url": "https://we.tl/s-random-hash",
}
`;
